kimu/core/intersection_tool_line_circle.py

```diff
@@ -55,16 +57,72 @@ def canvas_press_event(self, event: QgsMapToolEmitPoint) -> None:
+            - decimal.Decimal('2.0') * line_coords[0] * line_coords[2] \
+            + (line_coords[0]) ** decimal.Decimal('2.0')
+        b = -decimal.Decimal('2.0') * (line_coords[3]) ** decimal.Decimal('2.0') * line_coords[0] \
+            + decimal.Decimal('2.0') * line_coords[1] * line_coords[3] * line_coords[2] \
+            + decimal.Decimal('2.0') * line_coords[1] * line_coords[3] * line_coords[0] \
+            - decimal.Decimal('2.0') * (line_coords[1]) ** decimal.Decimal('2.0') * line_coords[2] \
+            - decimal.Decimal('2.0') * centroid[0] * (line_coords[2]) ** decimal.Decimal('2.0') \
+            - decimal.Decimal('2.0') * centroid[0] * (line_coords[0]) ** decimal.Decimal('2.0') \
+            + decimal.Decimal('4.0') * centroid[0] * line_coords[0] * line_coords[2] \
+            - decimal.Decimal('2.0') * line_coords[2] * centroid[1] * line_coords[3] \
+            + decimal.Decimal('2.0') * centroid[1] * line_coords[1] * line_coords[2] \
+            + decimal.Decimal('2.0') * centroid[1] * line_coords[3] * line_coords[0] \
+            - decimal.Decimal('2.0') * centroid[1] * line_coords[1] * line_coords[0]
+        c = (line_coords[3]) ** decimal.Decimal('2.0') * (line_coords[0]) ** decimal.Decimal('2.0') \
+            - decimal.Decimal('2.0') * line_coords[0] * line_coords[1] * line_coords[2] * line_coords[3] \
+            + (line_coords[1]) ** decimal.Decimal('2.0') * (line_coords[2]) ** decimal.Decimal('2.0') \
+            + (centroid[0]) ** decimal.Decimal('2.0') * (line_coords[2]) ** decimal.Decimal('2.0') \
+            - decimal.Decimal('2.0') * (centroid[0]) ** decimal.Decimal('2.0') * line_coords[0] * line_coords[2] \
+            + (centroid[0]) ** decimal.Decimal('2.0') * (line_coords[0]) ** decimal.Decimal('2.0') \
+            + decimal.Decimal('2.0') * line_coords[2] * centroid[1] * line_coords[3] * line_coords[0] \
+            - decimal.Decimal('2.0') * (line_coords[2]) ** decimal.Decimal('2.0') * centroid[1] * line_coords[1] \
+            - decimal.Decimal('2.0') * (line_coords[0]) ** decimal.Decimal('2.0') * centroid[1] * line_coords[3] \
+            + decimal.Decimal('2.0') * centroid[1] * line_coords[1] * line_coords[2] * line_coords[0] \
+            + (line_coords[2]) ** decimal.Decimal('2.0') * (centroid[1]) ** decimal.Decimal('2.0') \
+            - (line_coords[2]) ** decimal.Decimal('2.0') * r ** decimal.Decimal('2.0') \
+            - decimal.Decimal('2.0') * line_coords[0] * line_coords[2] * (centroid[1]) ** decimal.Decimal('2.0') \
+            + decimal.Decimal('2.0') * line_coords[0] * line_coords[2] * r ** decimal.Decimal('2.0') \
+            + (line_coords[0]) ** decimal.Decimal('2.0') * (centroid[1]) ** decimal.Decimal('2.0') \
+            - (line_coords[0]) ** decimal.Decimal('2.0') * r ** decimal.Decimal('2.0')
+        result = [a, b, c]
+        return result
+
+    def _one_layer(self, x_sol1: QVariant.Double, y_sol1: QVariant.Double) -> QgsVectorLayer:
```

Comment:
This and _two_layers could be more descriptive. Both seem to return only a single layer? So maybe something like a very simple `_create_layer` method that wraps these two cases and contains the if-else logic, something like this:
```
def _create_layer(..., sqrt_in) -> QgsVectorLayer:
    if float(sqrt_in) == 0.0: return self._create_single_intersection_layer(...)
    else: return self._create_multiple_intersection_layer(...)
```
or however you best see fit!

Reply:
I combined these functions into one since, after all, these functions only meaning was to generate result layer(s). By testing whether x_sol1 and x_sol2 variables were equal, removing a bunch of duplicate code was possible. I.e. if the two variables are equal -> return and else generate another result layer for intersection point 2.